### ISSUE TITLE
A11y: Android Voice Over has no access to Flyout detail page

### ISSUE THREAD

**Issue by jantanis** [xamarin/Xamarin.Forms#13694]
### Description
When using Voice Over on Android you cannot access elements on the Flyout detail page.

### Steps to Reproduce

1. Open sample solution attached
2. Run on Android with Voice Over on
3. Try to select someting on the detail page

### Expected Behavior
Detail page get's focus on push. This is also someting that could be improved on iOS as it does not get the focus.

### Actual Behavior
Detail page not even accessible.

### Basic Information

- Version with issue: 5.0.0.1931
- Last known good version: n/a
- Platform Target Frameworks: 
  - iOS:  12.4
  - Android: 26.1.1
- Android Support Library / AndroidX Version: 
- NuGet Packages:
- Affected Devices:

### Environment
<details><summary>Show/Hide Visual Studio info</summary>

```
=== Visual Studio Enterprise 2019 for Mac ===

Version 8.8.7 (build 18)
Installation UUID: c07e4a0f-0c5e-499c-ae74-6fa62671908d
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000113

=== Mono Framework MDK ===

Runtime:
	Mono 6.12.0.113 (2020-02/4fdfb5b1fd5) (64-bit)
	Package version: 612000113

=== Roslyn (Language Service) ===

3.8.0-5.20519.18+4c195c3ac1974edcefa76774d7a59a2350ec55fa

=== NuGet ===

Version: 5.8.0.6860

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/5.0.102/Sdks
SDK Versions:
	5.0.102
	5.0.101
	5.0.100
	3.1.405
	3.1.404
	3.1.403
	3.1.402
	3.1.401
	3.1.302
	3.1.301
	3.1.300
	3.1.202
	3.1.200
	3.1.102
	3.1.101
	3.1.100
	3.0.101
	3.0.100
	2.2.100
	2.1.810
	2.1.809
	2.1.701
	2.1.700
	2.1.505
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	5.0.2
	5.0.1
	5.0.0
	3.1.11
	3.1.10
	3.1.9
	3.1.8
	3.1.7
	3.1.6
	3.1.5
	3.1.4
	3.1.2
	3.1.1
	3.1.0
	3.0.1
	3.0.0
	2.2.0
	2.1.23
	2.1.22
	2.1.21
	2.1.20
	2.1.19
	2.1.18
	2.1.17
	2.1.16
	2.1.15
	2.1.14
	2.1.13
	2.1.12
	2.1.11
	2.1.9

=== .NET Core 3.1 SDK ===

SDK: 3.1.405

=== Xamarin.Profiler ===

Version: 1.6.12.29
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Apple Developer Tools ===

Xcode 12.4 (17801)
Build 12D4e

=== Xamarin.Mac ===

Xamarin.Mac not installed. Can't find /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/Version.

=== Xamarin.iOS ===

Version: 14.8.0.3 (Visual Studio Enterprise)
Hash: c51fabee8
Branch: xcode12.3
Build date: 2020-12-10 21:05:05-0500

=== Xamarin Designer ===

Version: 16.8.0.510
Hash: 44e3f3ce9
Branch: remotes/origin/d16-8
Build date: 2020-12-10 00:06:14 UTC

=== Xamarin.Android ===

Version: 11.1.0.26 (Visual Studio Enterprise)
Commit: xamarin-android/d16-8/a36ce73
Android SDK: /Users/jan/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 30.0.5
SDK Build Tools Version: 29.0.2

Build Information: 
Mono: 5e9cb6d
Java.Interop: xamarin/java.interop/d16-8@79d9533
ProGuard: Guardsquare/proguard/proguard6.2.2@ebe9000
SQLite: xamarin/sqlite/3.32.1@1a3276b
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-8@2fb1cbc

=== Microsoft OpenJDK for Mobile ===

Java SDK: /Users/jan/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 16.8.0.32
Hash: 01a7774
Branch: remotes/origin/d16-8
Build date: 2021-01-14 00:34:58 UTC

=== Android Device Manager ===

Version: 16.8.0.46
Hash: 0a81419
Branch: remotes/origin/d16-8
Build date: 2021-01-14 00:35:22 UTC

=== Build Information ===

Release ID: 808070018
Git revision: 0f6def3c0ff5f9cee230e65718561cb41c8eedc0
Build date: 2021-01-29 08:24:09-05
Build branch: release-8.8
Xamarin extensions: 0f6def3c0ff5f9cee230e65718561cb41c8eedc0

=== Operating System ===

Mac OS X 10.15.7
Darwin 19.6.0 Darwin Kernel Version 19.6.0
    Thu Oct 29 22:56:45 PDT 2020
    root:xnu-6153.141.2.2~1/RELEASE_X86_64 x86_64

=== Enabled user installed extensions ===

DeepClean 1.2.5
MFractor 4.4.6
Straight8's SpecFlow Integration 1.11.3.0
NuGet Package Management Extensions 0.24
AapIdeExtension.Mac 1.1.64


```

</details>

### Reproduction Link
[A11y.FlyoutPage.zip](https://github.com/xamarin/Xamarin.Forms/files/5942767/A11y.FlyoutPage.zip)

**Comment by suggyd** [comment]
@jantanis is your example using Shell navigation or just a Flyout page with a MenuPage derived from ContentPage as the Flyout and a Navigation page as the Detail?  I'm using the later i.e. not using Shell and I'm not having a problem with the flyout page (menu) getting focus on Android, it's working fine.  On iOS I am experiencing the problem that the focus goes to the status bar instead of to the Flyout page (menu)

**Comment by jantanis** [comment]
@suggyd  It is a regular flyout page. The problem is manifesting on tablets.

**Comment by suggyd** [comment]
@jantanis here is some code to set the focus in iOS, you just pass it your forms element, you'll need to inject it as a platform specific service:

```
public bool TrySetFocus(VisualElement target)
{
    if (target == null)
    {
        throw new ArgumentNullException(nameof(target));
    }

    var renderer = Xamarin.Forms.Platform.iOS.Platform.GetRenderer(target);

    if (renderer != null && renderer.NativeView != null)
    {
        UIAccessibility.PostNotification(UIAccessibilityPostNotification.ScreenChanged, renderer.NativeView);

        return true;
    }

    return false;
}

```
I inject this as a platform specific dependency, I have a version for android too if you need that? 

On my Flyout page I subscribe to the IsPresentedChanged event then use it to notify the menu page to set the focus:

```
private void Flyout_IsPresentedChanged(object sender, EventArgs e)
{
    if (IsPresented)
    {
        ((MenuPage)Flyout).OnPresented();
    }
}

```
In my Menu page OnPresented() I then use my AccessibilityFocusManager to set the focus to the first menu item.  This is just using the code above for iOS.

```
 public void OnPresented()
  {
      try
      {
          if (Device.RuntimePlatform == Device.iOS)
          {
              // I had to wrap this in an BeginInvokeOnMainThread to queue it behind the flyout actually being shown because
             // IsPresentedChanged gets called before the Flyout is visible.
              Device.BeginInvokeOnMainThread(()=>Xm.AccessibilityFocusManager.Current.TrySetFocus(settingsMenuButton));
          }
      }
      catch(Exception ex)
      {
        ...
      }
  }
```

**Comment by rachelkang** [comment]
Hi all! If you haven't already, I recommend updating to the latest version Xamarin.Forms or at least to Xamarin.Forms 5.0.0.2083, which has many accessibility improvements. These improvements include an experimental accessibility flag that addresses a lot of improvements in Android accessibility, including this one.

To leverage this flag on the latest versions of Xamarin Forms, the following code can be inserted in the App.xaml.cs file, following the InitializeComponent call and before the MainPage is set: `Xamarin.Forms.Device.SetFlags(new List<string> { "Accessibility_Experimental"});`

Let us know how that works for you!
